util/src/main/java/com/github/twitch4j/util/EnumUtil.java

```diff
@@ -0,0 +1,62 @@
+package com.github.twitch4j.util;
+
+import lombok.experimental.UtilityClass;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@UtilityClass
+public class EnumUtil {
```

Comment:
Note: follow-up PR will refactor other enums (with `MAPPINGS`) to use this utility class